### PR TITLE
runes: fix `checkrune` when `method` parameter is the empty string.

### LINKED ISSUE #6725: checkrune: when we pass an empty value for `method` parameter, it seems that `checkrune` return `{"valid": true}`
### Issue and Steps to Reproduce

Hey, I have question about `checkrune` behavior.

When we pass an empty value for `method` parameter, it seems that `checkrune` returns `{"valid": true}` even if the rune doesn't allow `getinfo` method as we can see in the following example.

With `checkrune`, first `method=getinfo` and then `method=invoice`:

    $ lightning-cli createrune null '[["method/getinfo"]]'
    {
       "rune": "19gTcT3uCrSVd8e7o2yb-GFr8Yy0P-61n8jijaF4FCc9MCZtZXRob2QvZ2V0aW5mbw==",
       "unique_id": "0"
    }
    $ lightning-cli -k checkrune rune=19gTcT3uCrSVd8e7o2yb-GFr8Yy0P-61n8jijaF4FCc9MCZtZXRob2QvZ2V0aW5mbw== method=getinfo
    {
       "code": 1502,
       "message": "Not permitted: method is equal to getinfo"
    }
    $ lightning-cli -k checkrune rune=19gTcT3uCrSVd8e7o2yb-GFr8Yy0P-61n8jijaF4FCc9MCZtZXRob2QvZ2V0aW5mbw== method=invoice
    {
       "valid": true
    }

And with `method` being empty we get:

    $ lightning-cli -k checkrune rune=19gTcT3uCrSVd8e7o2yb-GFr8Yy0P-61n8jijaF4FCc9MCZtZXRob2QvZ2V0aW5mbw== method=
    {
       "valid": true
    }

Is this the expected behavior of `checkrune`?

### `getinfo` output

```
{
   "id": "03b9dc9b130fe1ac20dc8a7f208b58409c87d0b0d3bf15e135ac23e365cc177f12",
   "alias": "REDGENESIS",
   "color": "03b9dc",
   "num_peers": 0,
   "num_pending_channels": 0,
   "num_active_channels": 0,
   "num_inactive_channels": 0,
   "address": [
      {
         "type": "ipv6",
         "address": "...",
         "port": 19846
      }
   ],
   "binding": [
      {
         "type": "ipv6",
         "address": "::",
         "port": 19846
      },
      {
         "type": "ipv4",
         "address": "0.0.0.0",
         "port": 19846
      }
   ],
   "version": "v23.08.1",
   "blockheight": 1,
   "network": "regtest",
   "fees_collected_msat": 0,
   "lightning-dir": "/tmp/l3-regtest/regtest",
   "our_features": {
      "init": "08a0000a0269a2",
      "node": "88a0000a0269a2",
      "channel": "",
      "invoice": "02000002024100"
   }
}
```

### PR DESCRIPTION
**Pull request by tonyaldon**
This fix #6725.

### CODE REVIEW

File: lightningd/runes.c [hunk 1]
```diff
@@ -827,7 +827,11 @@ static struct command_result *json_checkrune(struct command *cmd,
 	cinfo.runes = cmd->ld->runes;
 	cinfo.peer = nodeid;
 	cinfo.buf = buffer;
-	cinfo.method = method;
+	if (method != NULL && streq(method, "")){
+		cinfo.method = NULL;
+	} else {
+		cinfo.method = method;
+	}
```
Comment (vincenzopalazzo) [inline lightningd/runes.c:834]:
```suggestion
	if (!method && streq(method, ""))
		cinfo.method = NULL;
	else
		cinfo.method = method;
```

**Review by vincenzopalazzo — COMMENTED** [review]
just a small style suggestion.

I am thinking about why this is not NULL, I should check the p_opt. If the value is not specified I expect that the value will be NULL and not empty

File: lightningd/runes.c (hunk 1 above)
Comment (tonyaldon) [reply lightningd/runes.c]: Done ;)

**Comment by tonyaldon** [comment]
> I am thinking about why this is not NULL, I should check the p_opt.  If the value is not specified I expect that the value will be NULL and not empty

The value is specified, we pass the empty string `""` as `method` parameter and this is the problem of the issue #6725.

You can check the test `test_rune_method_not_equal_and_method_empty` added in that PR.

`p_opt` is working as expected.

**Comment by vincenzopalazzo** [comment]
>The value is specified, we pass the empty string "" as method parameter and this is the problem of the issue https://github.com/ElementsProject/lightning/issues/6725.

* Vincent should read the issue :)

File: tests/test_runes.py [hunk 2]
```diff
@@ -594,6 +594,17 @@ def test_missing_method_or_nodeid(node_factory):
...
+    l1 = node_factory.get_node()
+    rune = l1.rpc.createrune(restrictions=[["method/getinfo"]])['rune']
+
+    with pytest.raises(RpcError, match='Not permitted: method is equal to getinfo'):
+        l1.rpc.checkrune(rune=rune, method='getinfo')
+    assert l1.rpc.checkrune(rune=rune, method='invoice')['valid'] is True
+    with pytest.raises(RpcError, match='Not permitted: method not present'):
+        l1.rpc.checkrune(rune=rune, method='')
+
 def test_invalid_restrictions(node_factory):
```
Comment (vincenzopalazzo) [inline tests/test_runes.py:608]: give another endline to fix the CI 

```
2023-10-10T18:07:28.4276716Z checking for egrep... /usr/bin/grep -E
2023-10-10T18:07:28.9992884Z checking for ANSI C header files... yes
2023-10-10T18:07:29.1313563Z checking for sys/types.h... yes
2023-10-10T18:07:29.3154822Z checking for sys/stat.h... yes
2023-10-10T18:07:29.4359691Z checking for stdlib.h... tests/test_runes.py:608:1: E302 expected 2 blank lines, found 1
2023-10-10T18:07:29.5113573Z yes
2023-10-10T18:07:29.5341322Z make: *** [Makefile:538: check-python-flake8] Error 1
2023-10-10T18:07:29.5342110Z make: *** Waiting for unfinished jobs....
2023-10-10T18:07:29.7553430Z checking for string.h... yes
```

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: tests/test_runes.py (hunk 2 above)
Comment (tonyaldon) [reply tests/test_runes.py]: Done!

I forgot to run `make check-python-flake8` before commiting :(

Comment (ShahanaFarooqui) [reply tests/test_runes.py]: Tested with f0f9bb23307c7894995cdb89c3ac762c3bf3796e.

**Comment by ShahanaFarooqui** [comment]
@tonyaldon Great catch. Thanks for the fix.

ACK f0f9bb23307c7894995cdb89c3ac762c3bf3796e.

**Review by ShahanaFarooqui — APPROVED** [review]

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/6759/commits/f0f9bb23307c7894995cdb89c3ac762c3bf3796e

**Pull request merged**